cpp/src/io/avro/reader_impl.cu

```diff
@@ -456,6 +457,20 @@ std::unique_ptr<table> reader::impl::read(int skip_rows, int num_rows,
     }
   }
 
+  // Return metadata
+  if (metadata_out) {
+    // Return column names (must match order of returned columns)
+    metadata_out->column_names.resize(selected_columns.size());
+    for (size_t i = 0; i < selected_columns.size(); i++) {
+      metadata_out->column_names[i] = selected_columns[i].second;
+    }
+    // Return user metadata
+    metadata_out->user_data.clear();
+    for (const auto& kv : _metadata->user_data) {
```

Comment:
Can the loop be replaced with:
```c++
  metadata_out->user_data.insert(_metadata->user_data.begin(), _metadata->user_data.end())
```

Reply:
I'll try it out, I don't think they're actually the same type even if the underlying layout (pair of strings) is the same...

Reply:
Oh, actually, they were in fact both `map<std::string, std::string>`, so I just replaced this with
`metadata_out->user_data = _metadata->user_data;`